pkg/common/util/validation_test.go

```diff
@@ -0,0 +1,89 @@
+package util
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestValidateSpiffeId(t *testing.T) {
```

Comment:
According with the SPIFFE ID Memo at https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md, the SPIFFE ID must not include a query or fragment component. E.g.: `spiffe://example.org/path/service?query=myquery` must be rejected as a valid SPIFFE ID.
Also, the userinfo and port parts of the authority component must not be set, and the : delimiter must not be present. E.g.: `spiffe://user:password@example.org:8080/path` must be rejected.

Reply:
done, added those cases and proper tests